File: Modules/AUTOAI/AddonsModules/API/Jokes.js
async function JokesGrabber(filters) {
    try {
        if (filters) {
            const jokeurl = await fetch('https://v2.jokeapi.dev/joke/Any?blacklistFlags=nsfw,religious,political,racist,sexist,explicit', {
                method: 'GET',
            }).then(res => res.json());
        
            var responsetext = [
                'Okay, heres a Joke for you',
                jokeurl.setup,
                jokeurl.delivery,
                'Hope you like that joke?'
            ];
        } else {
            const jokeurl = await fetch('https://v2.jokeapi.dev/joke/Any', {
                method: 'GET',
            }).then(res => res.json());
        
            var responsetext = [
                'Okay, heres a Joke for you',
                jokeurl.setup,
                jokeurl.delivery,
                'Hope you like that joke?'
            ];
        }
        return { resp: responsetext };
    } catch (error) {
        console.error('Error fetching time data:', error);
        return null;
    }
}

module.exports = {
    JokesGrabber
}

File: nekosuneai/skills/jokes.py
"""Jokes (ported from Jokes.js) - jokeapi.dev."""

from __future__ import annotations

import requests

from ..logutil import log


def get_joke(allow_explicit: bool = False) -> list[str]:
    url = "https://v2.jokeapi.dev/joke/Any"
    if not allow_explicit:
        url += "?blacklistFlags=nsfw,religious,political,racist,sexist,explicit"
    try:
        data = requests.get(url, timeout=15).json()
    except Exception as exc:
        log("[Jokes] fetch failed:", exc)
        return ["Sorry, I couldn't think of a joke right now."]

    if data.get("type") == "twopart":
        setup, delivery = data.get("setup", ""), data.get("delivery", "")
        return ["Okay, here's a joke for you.", setup, delivery, "Hope you liked that one!"]
    single = data.get("joke", "")
    return ["Okay, here's a joke for you.", single, "Hope you liked that one!"]
